src/init/features/hosting/github.ts

```diff
@@ -0,0 +1,517 @@
+import * as fs from "fs";
+import * as yaml from "js-yaml";
+import { safeLoad } from "js-yaml";
+import * as ora from "ora";
+import * as path from "path";
+import * as sodium from "tweetsodium";
+import { Setup } from "../..";
+import { login } from "../../../auth";
+import { dirExistsSync } from "../../../fsutils";
+import {
+  createServiceAccount,
+  createServiceAccountKey,
+  deleteServiceAccount,
+} from "../../../gcp/iam";
+import { addServiceAccountToRoles, firebaseRoles } from "../../../gcp/resourceManager";
+import * as logger from "../../../logger";
+import { prompt } from "../../../prompt";
+import { logBullet, logLabeledBullet, logSuccess } from "../../../utils";
+import api = require("../../../api");
+
+const GITHUB_DIR = `.github`;
+const WORKFLOW_DIR = `${GITHUB_DIR}/workflows`;
+
+const YML_RELATIVE_PATH_PULL_REQUEST = `${WORKFLOW_DIR}/firebase-hosting-pull-request.yml`;
+const YML_FULL_PATH_PULL_REQUEST = path.join(process.cwd(), YML_RELATIVE_PATH_PULL_REQUEST);
+
+const YML_RELATIVE_PATH_MERGE = `${WORKFLOW_DIR}/firebase-hosting-merge.yml`;
+const YML_FULL_PATH_MERGE = path.join(process.cwd(), YML_RELATIVE_PATH_MERGE);
+
+const GIT_CONFIG_PATH = path.join(process.cwd(), ".git/config");
+
+const CHECKOUT_GITHUB_ACTION_NAME = "actions/checkout@v2";
+const HOSTING_GITHUB_ACTION_NAME = "FirebaseExtended/action-hosting-deploy@v0";
+
+export async function initGitHub(setup: Setup, config: any, options: any) {
+  // GitHub Oauth
+  logger.info();
+  logBullet(
+    "Authorizing with GitHub to upload your service account to a GitHub repository's secrets store."
+  );
+
+  const ghAccessToken = await signInWithGitHub();
+
+  // Get GitHub user Details
+  const userDetails = await getGitHubUserDetails(ghAccessToken);
+  const ghUserName = userDetails.login;
+
+  logger.info();
+  logSuccess(`Success! Logged into GitHub as ${bold(ghUserName)}`);
+  logger.info();
+
+  // Prompt for repo and validate by getting the public key
+  const { repo, key, keyId } = await promptForRepo(setup, ghAccessToken);
+
+  const { default_branch: defaultBranch, id: repoId } = await getRepoDetails(repo, ghAccessToken);
+
+  // Valid secret names:
+  // https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#naming-your-secrets
+  const githubSecretName = `FIREBASE_SERVICE_ACCOUNT_${setup.projectId
+    .replace(/-/g, "_")
+    .toUpperCase()}`;
+
+  const serviceAccountName = `github-action-${repoId}`;
+
+  const serviceAccountJSON = await createServiceAccountAndKeyWithRetry(
+    setup,
+    repo,
+    serviceAccountName
+  );
+
+  logger.info();
+  logSuccess(
+    `Created service account ${bold(serviceAccountName)} with Firebase Hosting admin permissions.`
+  );
+
+  const spinnerSecrets = ora.default(`Uploading service account secrets to repository: ${repo}`);
+  spinnerSecrets.start();
+
+  const encryptedServiceAccountJSON = await encryptServiceAccountJSON(serviceAccountJSON, key);
```

Comment:
Please fix the warning here - don't await the value if it's not a promise'd value

Reply:
fixed